packages/components/src/components/overlay/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, MouseEvent } from 'react';
+import {
+  Drawer as ChakraDrawer,
+  DrawerOverlay as ChakraDrawerOverlay,
+  DrawerContent as ChakraDrawerContent,
+  DrawerFooter as ChakraDrawerFooter,
+  DrawerBody as ChakraDrawerBody,
+  DrawerHeader as ChakraDrawerHeader,
+  DrawerProps,
+  ModalOverlayProps,
+  DrawerContentProps,
+  ModalFooterProps,
+  ModalBodyProps,
+  ModalHeaderProps,
+  forwardRef,
+  useModalContext,
+} from '@chakra-ui/react';
+import { DrawerStaticMembers } from './types';
+import { CloseButton, CloseButtonProps } from '@components';
+import { callAllHandlers } from '@chakra-ui/utils';
+
+
+export const Drawer: FC<DrawerProps> & DrawerStaticMembers = ({ children, ...rest }) => (
+  <ChakraDrawer {...rest}>
+    {children}
+  </ChakraDrawer>
+);
+
+const DrawerOverlay: FC<ModalOverlayProps> = (props) => <ChakraDrawerOverlay {...props} />;
+
+const DrawerContent: FC<DrawerContentProps> = ({ children, ...rest }) => (
+  <ChakraDrawerContent {...rest}>
+    {children}
+  </ChakraDrawerContent>
+);
+
+const DrawerFooter: FC<ModalFooterProps> = ({ children, ...rest }) => (
+  <ChakraDrawerFooter {...rest}>
+    {children}
+  </ChakraDrawerFooter>
+
+);
+
+const DrawerBody: FC<ModalBodyProps> = ({ children, ...rest }) => (
+  <ChakraDrawerBody {...rest}>
+    {children}
+  </ChakraDrawerBody>
+);
+
+const DrawerCloseButton = forwardRef<CloseButtonProps, typeof CloseButton>((props, ref) => {
+  const { onClick, ...rest } = props;
+  const { onClose } = useModalContext();
```

Comment:
Can we use the `useDisclosure` hook instead?

Reply:
Didn't work :/